udf_js/unzip.sql

```diff
@@ -0,0 +1,53 @@
+/*
+
+Unzips a GZIP string.   
+
+This implementation relies on the zlib.js library (https://github.com/imaya/zlib.js) and 
+the atob function for decoding base64.
+
+*/
+
+
+CREATE TEMP FUNCTION
+  udf_js_unzip (input BYTES)
+  RETURNS STRING
+  LANGUAGE js AS """
+    // converts a byte array to a string
+    function binary2String(byteArray) {
+        return String.fromCharCode.apply(String, byteArray);
+    }
+    
+    // input is in base64, so it needs to be decoded
+    var decodedData = atob(input);
+    var compressedData = decodedData.split('').map(function(e) {
+        return e.charCodeAt(0);
+    });
+    
+    var gunzip = new Zlib.Gunzip(compressedData); 
```

Comment:
Will this raise an error if the data is not actually a gzip stream? We may want to catch that case and pass back raw bytes through binary2string instead in that case. Or perhaps we could have an additional `safe_` variant that catches exceptions and returns null.